backend/appserver/views.py

```diff
@@ -12,13 +15,81 @@ def health(request):
 
 def courses_list(request):
     if request.method == 'GET':
-        #courses = serializers.serialize('json', Courses.objects.all())
-        #courses = Courses.objects.all()
         title = request.GET.get('title', None)
-        print(title)
-        return JsonResponse({'title': title})
-        # return HttpResponse(courses)
-        #title = request.query_params.get('title', None)
-        # if title is not None:
-        #    courses = courses.filter(title_icontains=title)
-        # return JsonResponse({'title': courses})
+        udemy = udemy_courses(search_title=title, page=1, page_size=12)
+        edx = edx_courses(search_title=title, page=1, page_size=12)
+        courses = {'data': udemy+edx}
+        return JsonResponse(courses)
+
+
+def udemy_courses(search_title, page, page_size):
+    url = 'https://www.udemy.com/api-2.0/courses'
+    client_id = 'Y5JE1ujYlCvHJtCwaUbMg2FS5a7IkUbbnMpttNZw'
+    client_secret = 'tR4S9whpnnaUCJ93wf26OFlaYXgNobzDiox8IfLDonHuYUHotltl4xOAF5e071URoyJD3hokCnkQ49IvUwvnG6fphVPxLV75zLUB2ArcFQxM3w1DP2TnE6xkKZ0TqrCx'
+    auth = (client_id, client_secret)
+    headers = {
+        'Content-Type': 'application/json;charset=utf-8'
+    }
+    params = {
+        'page': page,
+        'page_size': page_size,
+        'search': search_title,
+        'fields[course]': 'title,headline,avg_rating,price,url,num_subscribers,image_50x50,visible_instructors',
+    }
+
+    response = requests.get(url, auth=auth, headers=headers, params=params)
+    courses = response.json()
+    search_title = search_title.lower()
+    data = []
+
+    for item in courses['results']:
+        if search_title in item['title'].lower():
+            # print(sort_data(item))
+            clean = sort_data(item, 'Udemy')
+            data.append(clean)
+    return data
+
+
+def edx_courses(search_title, page, page_size):
+    url = 'https://www.udemy.com/api-2.0/courses'
+    client_id = 'Y5JE1ujYlCvHJtCwaUbMg2FS5a7IkUbbnMpttNZw'
+    client_secret = 'tR4S9whpnnaUCJ93wf26OFlaYXgNobzDiox8IfLDonHuYUHotltl4xOAF5e071URoyJD3hokCnkQ49IvUwvnG6fphVPxLV75zLUB2ArcFQxM3w1DP2TnE6xkKZ0TqrCx'
+    auth = (client_id, client_secret)
+    headers = {
+        'Content-Type': 'application/json;charset=utf-8'
+    }
+    params = {
+        'page': page,
+        'page_size': page_size,
+        'search': search_title,
+        'fields[course]': 'title,headline,avg_rating,price,url,num_subscribers,image_50x50,visible_instructors',
+    }
+
+    response = requests.get(url, auth=auth, headers=headers, params=params)
+    courses = response.json()
+    search_title = search_title.lower()
+    data = []
+
+    for item in courses['results']:
+        if search_title in item['title'].lower():
+            clean = sort_data(item, 'Edx')
+            data.append(clean)
+    return data
+
+
+def sort_data(data, platform):
```

Comment:
Sort is a very defined term in API, it is used to signal when data is sorted (ordered) given a set of parameters, in this case, this function does not perform any sort operation, improve the name of this function